tests/unit_tests/job_runner/test_job.py

```diff
@@ -33,6 +33,47 @@ def test_run_with_process_failing(
         next(run)
 
 
+@pytest.mark.usefixtures("use_tmpdir")
+def test_memory_usage_counts_grandchildren():
+    scriptname = "recursive_memory_hog.py"
+    with open(scriptname, "w", encoding="utf-8") as script:
+        script.write(
+            """#!/usr/bin/env python
+import os
+import sys
+import time
+
+counter = int(sys.argv[1])
+numbers = list(range(int(1e6)))
+if counter > 0:
+    parent = os.fork()
```

Comment:
How this will work on Mac?

Reply:
I have no idea if this will work differently on Mac... Passing tests is noted :) (docs do not hint towards there being a mac differnce here..?)